src/nl/surf/eduhub_rio_mapper/cli.clj

```diff
@@ -178,6 +180,31 @@
       (assoc (if (= type "opleidingsrelatiesBijOpleidingseenheid") ::rio/opleidingscode ::ooapi/id) id
              ::rio/type type)))
 
+(defn- make-async-callback [req]
+  (async/thread
+    (loop [retries-left 3]
+      (when-not (http-status/success-status? (-> req http-utils/send-http-request :status))
+        (log/warnf "Could not reach webhook %s" (:url req))
+        (Thread/sleep 30000)
+        (when (pos? retries-left)
+          (recur (dec retries-left)))))))
+
+(defn make-set-status-fn [config]
+  ; data is result of run-job-fn, which is result of job/run!, which is result of update-and-mutate or delete-and-mutate
+  ; which is the result of the mutator/make-mutator, which is the result of handle-rio-mutate-response, which
+  ; is the parsed xml response converted to edn
+  (fn [{:keys [callback resource] :as job} status & [xml-resp]]
+    (let [data (-> xml-resp vals first)
+          code (:opleidingseenheidcode data)]
+      (status/set! config (:token job) status data)
+      (when (and callback (#{:done :error :time-out} status))
+        (make-async-callback {:url          callback
+                              :method       :post
+                              :content-type :json
+                              :body         (merge {:status     status
+                                                    :resource   resource}
+                                                   (when code {:attributes {:opleidingseenheidcode code}}))})))))
```

Comment:
Move request construction into `do-async-callback` because that's the function which should know about the "how" of callbacks, this function should just trigger it.